docs/reference/gc-toolkit/gc-scripts-hub/dataset-uploader.md

```diff
@@ -0,0 +1,115 @@
+---
+sidebar_position: 2
+---
+
+# GC Dataset Importer
+
+Upload a dataset file and turn it into a clean PostgreSQL table in your Guardian Connector data warehouse. The app guides you through naming the dataset, uploading a file, optionally tagging the **data source**, and finalizing the import. It also saves the original and transformed files to your deployment’s data lake (accessible via [Filebrowser](/reference/gc-toolkit/filebrowser/)).
+
+---
+
+## 📦 What this tool does
+
+* Converts and validates common **tabular** files (`.csv`, `.xlsx/.xls`, `.json`) to CSV for ingestion
+* Converts and validates common **spatial** files (`.geojson`, `.kml`, `.gpx`, `.kmz`) to GeoJSON for ingestion
+* Optionally applies **source-specific** transformations (e.g., KoboToolbox, CoMapeo, ODK, Locus Map)
+* Writes the result to a **new** Postgres table and archives files to file storage
+
+---
+
+## Step-by-Step: Import a Dataset
+
+### 1. Open the app in Windmill
+
+From **GC Scripts Hub → Apps**, click **GC Dataset Importer**.
+
+![Open the GC Dataset Importer app](./images/dataset-uploader-1.jpg)
+
+### 2. Name your dataset
+
+Give your dataset a clear, human-readable name, then click **Check Availability**.
+
+![Enter a dataset name](./images/dataset-uploader-2.jpg)
+
+If available, you’ll see the resulting table name you’ll get in the warehouse.
+
+![Name is available and mapped to table name](./images/dataset-uploader-3.jpg)
+
+### 3. Upload your file
+
+Select your file and click **Upload File**. When the upload completes, proceed to the next step.
+
+![Upload your file and confirm success](./images/dataset-uploader-4.jpg)
```

Comment:
Once you've uploaded the data, the upload button is still enabled.
Is there any use for that, or would it be better to disable it once uploaded?

Reply:
Huh, you're right. That shouldn't be the case actually. I'll need to fix that and adjust the screenshot.